stb_vorbis.c

```diff
@@ -3642,7 +3643,9 @@ static int start_decoder(vorb *f)
    f->vendor[len] = (char)'\0';
    //user comments
    f->comment_list_length = get32_packet(f);
-   f->comment_list = (char**)setup_malloc(f, sizeof(char*) * (f->comment_list_length));
+   f->comment_list = NULL;
+   if (f->comment_list_length > 0)
+       f->comment_list = (char**) setup_malloc(f, sizeof(char*) * (f->comment_list_length));
    if (f->comment_list == NULL)                     return error(f, VORBIS_outofmem);
```

Comment:
I don't think this works.
In 3646 you set it to NULL, afterwards you only set it to something else if length > 0, and here you still `return error(f, VORBIS_outofmem);` if it's NULL, so loading vorbis files with no comment will still fail.

Reply:
thanks for noticing that, i missed that, ill update the pr in a few